sample_script.ps1

```diff
@@ -143,12 +171,149 @@ function UploadData ($access_token) {
         throw "Service unavailable."
     }
     else {
-        WriteErrorMessage("Failure with StatusCode [{0}] and ReasonPhrase [{1}]" -f $result.StatusCode, $result.ReasonPhrase)
-        WriteErrorMessage("Error body : {0}" -f $result.Content.ReadAsStringAsync().Result)
+        Write-ErrorMessage("Failure with StatusCode [{0}] and ReasonPhrase [{1}]" -f $result.StatusCode, $result.ReasonPhrase)
+        Write-ErrorMessage("Error body : {0}" -f $result.Content.ReadAsStringAsync().Result)
     }
 }
 
-RetryCommand -ScriptBlock {
-    $access_token = GetAccessToken
-    UploadData($access_token)
+function GetOrCreateMetadata($FileName) {
+    $fileHash = ComputeHashForInputFile($FileName)
+    $metaDataFileName = $TmpDirName + "\." + $fileHash + "####mdata.txt"
```

Comment:
Create a helper function for creating this file name and constant string for suffix, being used at multiple places.